java/src/jmri/implementation/AbstractTurnout.java

```diff
@@ -129,13 +131,39 @@ public void setCommandedState(int s) {
     }
 
     /**
-     * Define duration of delay for DELAYED feedback mode.
+     * Duration in Milliseconds of delay for DELAYED feedback mode.
      * <p>
      * Defined as "public non-final"
      * so it can be changed in e.g. the jython/SetDefaultDelayedTurnoutDelay script.
      */
     public static int DELAYED_FEEDBACK_INTERVAL = 4000;
 
+    /**
+     * Duration in Milliseconds of interval between separate Turnout commands.
+     * <p>
+     * Read and change from e.g. XNetTurnout extensions and scripts using #setOutputInterval(int)
+     */
+    private int TURNOUT_INTERVAL = InstanceManager.turnoutManagerInstance().getInterval();
+
+    public void setOutputInterval(int newInterval) {
+        TURNOUT_INTERVAL = newInterval;
+        log.debug("(jmri.implementation.abstractTurnout_TURNOUT_INTERVAL set to: {}", newInterval);
+    }
+
+    public void waitOutputInterval() {
+        if (TURNOUT_INTERVAL > 0) { // is set in the Memo per hardware connection, default = 0
+            log.debug("interval = {} ms", TURNOUT_INTERVAL);
+            // insert wait before next output command
+            try {
+                Thread.sleep(TURNOUT_INTERVAL);
```

Comment:
We need to be careful because this sleep could cause problems with the user interface.
If I understand correctly, this function gets called from setCommandedState() of the various turnout implementations. That is by the threading model of JMRI always supposed to be called on the layout thread, which at present moment is the AWT event thread.
The implementation performs an inline sleep, which will cause the AWT event processing thread to hang for however long the interval is set; say 200 msec. During that time there will be no event processing happening, so all the windows will appear unresponsive to user actions.
This problem will get a lot worse if there are multiple turnouts being set; for example a logix that sets a dozen turnouts could freeze the user interface for two seconds.
We also need to check that this function is never called during the processing of an incoming message. If someone else on the layout sets a turnouts and that incoming message causes the delay function to be called, then we are limited to being able to process only 5 messages coming in per second. Most layout control buses have the ability to deliver hundreds of messages to JMRI per second.
There is a third possible problem related to TurnoutControlledSignalHead and blinking aspects. Those are automatically generating turnout set commands as the output is switched on and off. With more than 5 blinking lights on the layout the user interface will be permanently locked up.

Generally instead of sleep, one could use ThreadingUtil.runOnLayoutDelayed(...), which is much better because that would immediately free up the AWT thread for processing user events. However, that will only guarantee that the setting happens 200 msec later than the setCommandedState call, but not that it will be 200 msec after the last outgoing turnout set command.